src/app/payment-options/new-payment-option/new-payment-option.component.ts

```diff
@@ -14,16 +15,21 @@ declare var $: any;
     styleUrls: ['./new-payment-option.component.css']
 })
 export class NewPaymentOptionComponent implements OnInit, AfterViewInit {
-    creditCardNavTab: JQuery;
-    bankAccountNavTab: JQuery;
-
     hasNoBankAccounts: boolean;
-
     bankCodes: BankCodeModel[];
 
+    bankAccountForm: FormGroup;
+
     constructor(private paymentService: PaymentService,
                 private router: Router,
-                private route: ActivatedRoute) {
+                private route: ActivatedRoute,
+                private fb: FormBuilder) {
+
+        this.bankAccountForm = this.fb.group({
```

Comment:
The task was misunderstood. The user **should be able to enter IBAN with spaces** - the app should remove the spaces before sending it to the server. No need for validation.

Reply:
Fixed!